ignite/metrics/gan/inception_score.py

```diff
@@ -0,0 +1,110 @@
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            from ignite.metric.gan.IS import InceptionScore
+            import torch
+
+            probabilities = torch.rand(10, 1000)
+
+            m = InceptionScore(num_probabilities=1000)
+            m.update(probabilities)
+            print(m.compute())
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        num_probabilities: Optional[int] = None,
+        prediction_model: Optional[Callable] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = "cpu",
+    ) -> None:
+
+        if num_probabilities is None and prediction_model is None:
+            num_probabilities = 1000
+            prediction_model = InceptionModel(return_features=False)
+        elif num_probabilities is None:
+            raise ValueError("Argument num_probabilities should be defined")
+        elif prediction_model is None:
+            self._prediction_model = lambda x: x
+            prediction_model = self._prediction_model
+
+        if num_probabilities <= 0:
+            raise ValueError(f"Argument num_probabilities must be greater to zero, got: {num_probabilities}")
+        self._num_probs = num_probabilities
+        self._eps = 1e-16
+        super(InceptionScore, self).__init__(output_transform=output_transform, device=device)
+
+    def _check_feature_input(self, samples: torch.Tensor) -> None:
+        if samples.dim() != 2:
+            raise ValueError(f"Probabilities must be a tensor of dim 2, got: {samples.dim()}")
+        if samples.shape[0] == 0:
+            raise ValueError(f"Batch size should be greater than one, got: {samples.shape[0]}")
+        if samples.shape[1] != self._num_probs:
+            raise ValueError(f"Number of Probabilities should be {self._num_probs}, got: {samples.shape[1]}")
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._num_examples = 0
+        self._prob_total = torch.zeros(self._num_probs, dtype=torch.float64).to(self._device)
+        self._total_kl_d = torch.zeros(self._num_probs, dtype=torch.float64).to(self._device)
+        super(InceptionScore, self).reset()
+
+    @reinit__is_reduced
+    def update(self, samples: torch.Tensor) -> None:
+        probabilities = self._prediction_model(samples.detach()).to(self._device)
```

Comment:
- use `with torch.no_grad()` instead of detaching.
- what about device mismatch here if samples is on cuda ?

Same remarks for FID !